spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -593,14 +599,31 @@ private void validateKey(Object entity, PathElement ancestorPE) {
 			BaseEntity entity, T convertedObject, ReadContext context) {
 		datastorePersistentEntity.doWithAssociations(
 				(AssociationHandler) (association) -> {
-					DatastorePersistentProperty referencePersistentProperty = (DatastorePersistentProperty) association
+					DatastorePersistentProperty referenceProperty = (DatastorePersistentProperty) association
 							.getInverse();
-					Object referenced = findReferenced(entity, referencePersistentProperty, context);
-					if (referenced != null) {
-						datastorePersistentEntity.getPropertyAccessor(convertedObject)
-								.setProperty(referencePersistentProperty, referenced);
+					String fieldName = referenceProperty.getFieldName();
+					if (entity.contains(fieldName) && !entity.isNull(fieldName)) {
```

Comment:
Looks like this condition checking modifies the behavior from before? I.e. if the `entity.contains(fieldName) && !entity.isNull(fieldName)` fails it will not execute the `referenced = findReferenced(entity, referenceProperty, context);` whereas before it did not check this condition.